### PR TITLE
Fixes #71 build tool sometimes places huts the wrong way

### LINKED ISSUE #71: Build tool sometimes places huts facing into the wrong direction.
### Minecolonies version
Newest
### Expected behavior
Rotate and place the hut facing the wanted direction.
### Actual behaviour
Sometimes faces opposite direction.
### Steps to reproduce the problem
Place a hut using the build tool, probably citizen hut at east or west will work.

### PR DESCRIPTION
**Pull request by Kostronor**
Closes #71 

# Changes proposed in this pull request:
- Fixes the build tool placing the huts the wrong way
- Fixes the build tool achievement.

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/items/ItemBuildTool.java [hunk 1]
```diff
@@ -27,13 +27,12 @@ public ItemBuildTool()
     @Override
     public EnumActionResult onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
     {
-        if (!worldIn.isRemote)
+        if (worldIn.isRemote)
         {
-            return EnumActionResult.FAIL;
+            playerIn.addStat(ModAchievements.achievementWandOfbuilding);
```
Comment (Kostronor) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: If we move the achievement gain inside a `worldIn.isRemote` then on a remote server the achievement won't be awarded.

Comment (Kostronor) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: Can you test that the achievement gets awarded on build tool use?

Comment (DarVillous) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: It did not, after looking at the code I'm not sure it ever did as the server would have exited the method before it was called.  I moved it to the top of the method to make sure it is called on use.

**Comment by Kostronor** [comment]
LGTM! @Raycoms @cltnschlosser what do you think?

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/minecolonies/items/ItemBuildTool.java [hunk 2]
```diff
@@ -27,14 +27,13 @@ public ItemBuildTool()
...
+        if (worldIn.isRemote)
         {
-            return EnumActionResult.FAIL;
+            MineColonies.proxy.openBuildToolWindow(pos.offset(facing));
         }
 
-        playerIn.addStat(ModAchievements.achievementWandOfbuilding);
-        MineColonies.proxy.openBuildToolWindow(pos.offset(facing));
-        return EnumActionResult.FAIL;
+        return EnumActionResult.SUCCESS;
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: What is the difference between FAIL and SUCCESS here?

Comment (Raycoms) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: Previously the itemUse method returned true and false, now it's fail and success.

Comment (cltnschlosser) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: But what does a FAIL and a SUCCESS represent? I remember it not being always just return true if the item was used, the return value had impact

Comment (Kostronor) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: Success means we have worked on that interact and it is done, so forge knows that something worked. A fail is used for not allowed stuff. and pass is used for spectating events.

**Pull request merged**